File: 0501~1000/0921/Minimum Add to Make Parentheses Valid.swift
class Solution {
    func minAddToMakeValid(_ s: String) -> Int {
        var arr = Array(s), cnt : Int = 0, ans : Int = 0
        for ch in s{
            if ch == "("{
                cnt += 1
            }
            else if cnt == 0{
                ans += 1
            }
            else{
                cnt -= 1
            }
        }
        return ans + cnt
    }
}

File: 0501~1000/0921/Minimum Add to Make Parentheses Valid.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        cnt = 0; ans = 0
        for ch in s:
            if ch == '(':
                cnt += 1
            elif cnt == 0:
                ans += 1
            else:
                cnt -= 1
        return ans + cnt
        
